grype/vulnerability/namespace.go

```diff
@@ -37,20 +37,27 @@ func githubJavaNamer(p *pkg.Package) []string {
 // TODO: expand with namer mapping to be more generic?
 func distroNamespace(d distro.Distro) string {
 	// TODO: can we drive this from information from grype-db? that would be ideal...
-	var distroStr string
-	switch d.Type {
-	case distro.CentOS, distro.RedHat:
-		distroStr = "rhel"
-	case distro.AmazonLinux:
-		distroStr = "amzn"
-	case distro.OracleLinux:
-		distroStr = "ol"
-	case distro.Alpine:
-		// XXX this assumes that a major and minor versions will always exist in Segments
-		segments := d.Version.Segments()
-		return fmt.Sprintf("%s:%d.%d", d.Type.String(), segments[0], segments[1])
-	default:
-		distroStr = d.Type.String()
+	var distroStr = d.Type.String()
+
+	var versionSegments []int
+	if d.Version != nil {
+		versionSegments = d.Version.Segments()
+	}
+
+	if len(versionSegments) > 0 {
+		switch d.Type {
+		// derived from https://github.com/anchore/anchore-engine/blob/5bbbe6b9744f2fb806198ae5d6f0cfe3b367fd9d/anchore_engine/services/policy_engine/__init__.py#L149-L159
+		case distro.CentOS, distro.RedHat, distro.Fedora:
+			// TODO: there is no mapping of fedora version to RHEL latest version (only the name)
+			return fmt.Sprintf("rhel:%d", versionSegments[0])
```

Comment:
I like the new shape of this flow, with returning directly once the needed values are known!

Reply:
given the variations, this did seems a little easier to read 👍